llmstack/datasources/handlers/databases/mysql.py

```diff
@@ -0,0 +1,202 @@
+import json
+import logging
+from typing import Dict, List, Optional
+
+from pydantic import Field
+
+from llmstack.common.blocks.base.schema import BaseSchema as _Schema
+from llmstack.common.blocks.data.store.database.database_reader import (
+    DatabaseReader,
+    DatabaseReaderInput,
+)
+from llmstack.common.blocks.data.store.database.mysql import MySQLConfiguration
+from llmstack.common.blocks.data.store.database.sqlalchemy import DatabaseType
+from llmstack.common.blocks.data.store.vectorstore import Document
+from llmstack.common.utils.models import Config
+from llmstack.datasources.handlers.datasource_processor import (
+    DataSourceEntryItem,
+    DataSourceProcessor,
+    DataSourceSchema,
+)
+from llmstack.datasources.models import DataSource
+
+logger = logging.getLogger(__name__)
+
+
+class MySQLConnection(_Schema):
+    host: str = Field(description="Host of the MySQL instance")
+    port: int = Field(
+        description="Port number to connect to the MySQL instance",
+    )
+    database_name: str = Field(description="MySQL database name")
+    username: str = Field(description="MySQL username")
+    password: Optional[str] = Field(description="MySQL password")
+
+
+class MySQLDatabaseSchema(DataSourceSchema):
+    connection: Optional[MySQLConnection] = Field(
+        description="MySQL connection details",
+    )
+
+
+class MySQLConnectionConfiguration(Config):
+    config_type = "mysql_connection"
+    is_encrypted = True
+    mysql_config: Optional[Dict]
+
+
+class MySQLDataSource(DataSourceProcessor[MySQLDatabaseSchema]):
```

Comment:
Instead of having a handler per engine, can have do a single SQLDatasource instead. Use union to support various sql engine ?